src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/RoutePatternUsageDetector.cs

```diff
@@ -0,0 +1,200 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+
+internal record struct RoutePatternUsageContext(
+    IMethodSymbol? MethodSymbol,
+    bool IsMinimal,
+    bool IsMvcAttribute);
+
+internal static class RoutePatternUsageDetector
```

Comment:
the EA layer exposes AspNetCoreCSharpRouteSyntaxDetector now.  Do you need this type anymore?

Reply:
hrmm... looks like you might, since you're looking for lots of special methods.